private/projects/cdk-graph-plugin-diagram.ts

```diff
@@ -114,4 +117,16 @@ export class CdkGraphPluginDiagramProject extends CdkGraphPluginProject {
       this.addGitIgnore(".jest-image-snapshot-touched-files");
     }
   }
+
+  private executeInWorkspace() {
+    switch (this.package.packageManager) {
+      case NodePackageManager.YARN:
+      case NodePackageManager.YARN2:
+        return "yarn workspace";
+      case NodePackageManager.PNPM:
+        return "pnpm --filter";
+      default:
+        return "npx -p";
+    }
+  }
```

Comment:
Similar comment as `execute` on NxMonorepo, this is verb and should be renamed. Additionally consider support taking in command to append for cleaner implementation usage.

Should we actually add this functionality to PDKProject so that all subclass, such as this, can utilize this common functionatlity? I would make it public on PDKProject as pretty core functionality that is beneficial.